jbpm-system-test/jbpm-timer-migration-test/src/test/java/org/kie/samples/integration/TimerMigrationSystemTest.java

```diff
@@ -0,0 +1,160 @@
+        args.put("SERVER", System.getProperty("org.kie.samples.server"));
+    }
+    
+    public static Network network = Network.newNetwork();
+    
+    @Container
+    public static MySQLContainer<?> mySQLContainer = new MySQLContainer<>(System.getProperty("org.kie.samples.image.mysql","mysql:latest"))
+                                        .withDatabaseName("bpms64kie")
+                                        .withUsername("rhpamuser")
+                                        .withPassword("rhpampassword")
+                                        .withFileSystemBind("src/test/resources/mysql", "/docker-entrypoint-initdb.d",
+                                                            BindMode.READ_ONLY)
+                                        .withNetwork(network)
+                                        .withNetworkAliases("mysql");
+    
+    public static KieServerContainer kieServer = new KieServerContainer(network, args);
+    
+    private static KieServicesClient ksClient;
+    
+    private static ProcessServicesClient processClient;
+    private static ProcessAdminServicesClient processAdminClient;
+    
+    @BeforeAll
+    public static void setup() {
+        kieServer.start();
+
+        logger.info("KIE Server started at port "+kieServer.getKiePort());
+        logger.info("mysql started at "+mySQLContainer.getJdbcUrl());
+
+        ksClient = authenticate(kieServer.getKiePort(), DEFAULT_USER, DEFAULT_PASSWORD);
+
+        processClient = ksClient.getServicesClient(ProcessServicesClient.class);
+        processAdminClient = ksClient.getServicesClient(ProcessAdminServicesClient.class);
+    }
+    
+    @BeforeEach
+    public void before() {
+        createContainer(ksClient);
+    }
+
+    @AfterAll
+    public static void tearDown() throws Exception {
+        kieServer.stop();
+        
+        DockerClient docker = DockerClientFactory.instance().client();
+        docker.listImagesCmd().withLabelFilter("autodelete=true").exec().stream()
+         .filter(c -> c.getId() != null)
+         .forEach(c -> docker.removeImageCmd(c.getId()).withForce(true).exec());
+    }
+
+    @Test
+    void testEJBTimerWithRollback() throws InterruptedException, SQLException {
+        Long processInstanceId = processClient.startProcess(containerId, "case03200412.timertest", Collections.singletonMap("timerAlert","2022-10-10T20:00:00.000+02:00"));
+        
+        assertTrue(processInstanceId>0);
+        
+        logger.info("Sending signal");
+        processClient.signal(containerId, "proceed", null);
+        
+        List<TimerInstance> timers = processAdminClient.getTimerInstances(containerId, processInstanceId);
+        
+        assertEquals(1, timers.size());
+        
+        timers = processAdminClient.getTimerInstances(containerId, 6L);
+        
+        logger.info("Retrieving timers for process 6; timers[6]: "+timers);
+        
+        assertEquals(1, timers.size());
```

Comment:
as already discussed, not sure about the migration process, since we are loading a mysql by using jbpm schemas version 6, and then we create a process instance through a kie-server 7.x? shouldn't we migrate the instance first? or am I missing anything here?